spec/spec_helper.rb

```diff
@@ -37,29 +38,26 @@ def puts(payload)
   c.include LogStashHelper
   c.extend LogStashHelper
 
-  c.before(:each) do
-    # TODO: commented out on post-merged in master - the logger has moved to log4j
-    #
-    #
-    # Force Cabin to always have a JSON subscriber.  The main purpose of this
-    # is to catch crashes in json serialization for our logs. JSONIOThingy
-    # exists to validate taht what LogStash::Logging::JSON emits is always
-    # valid JSON.
-    # jsonvalidator = JSONIOThingy.new
-    # allow(Cabin::Channel).to receive(:new).and_wrap_original do |m, *args|
-    #   logger = m.call(*args)
-    #   logger.level = :debug
-    #   logger.subscribe(LogStash::Logging::JSON.new(jsonvalidator))
-    #
-    #   logger
-    # end
+  # Some tests mess with global application state; ensure each spec runs in isolation
+  spec_mutex = Mutex.new
```

Comment:
@yaauie I think this is't necessary. RSpec executes tests sequentially, I don't think this changes behaviour at all?

Reply:
IIRC, the default spec runner is sequential, which means that this is in fact a no-op. I can remove it :)